x-pack/plugins/ingest_manager/server/plugin.ts

```diff
@@ -148,26 +133,26 @@ export class IngestManagerPlugin implements Plugin<IngestManagerSetupContract> {
         basePath: core.http.basePath,
       });
     }
-    return deepFreeze({
-      esIndexPatternService: new ESIndexPatternSavedObjectService(),
-      agentService: {
-        getAgentStatusById,
-      },
-    });
   }
 
   public async start(
     core: CoreStart,
     plugins: {
       encryptedSavedObjects: EncryptedSavedObjectsPluginStart;
     }
-  ) {
+  ): Promise<RecursiveReadonly<IngestManagerStartupContract>> {
```

Comment:
:information_source: Never seen `RecursiveReadOnly` - what's it do?

Reply:
It marks all properties as readonly traversing nested objects including collections.